File: 關聯法則/關聯法則.r
#read
#利用read.transactions將所有產品整理
library(arules)
data = read.transactions('Market_Basket_Optimisation.csv',
                         sep = ',', rm.duplicates = TRUE)
#查看購買資訊
summary(data)
#畫出前n名的產品
itemFrequencyPlot(data, topN = 10)

#利用先驗演算法運算
#parameter設定支持度及信心度
rules = apriori(data, parameter = list(support = 0.003, confidence = 0.2))

#視覺化(以提升度排序)
inspect(sort(rules, by = 'lift')[1:10])

File: 關聯法則/關聯法則.py
#套件
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#read
data = pd.read_csv('Market_Basket_Optimisation.csv',
                   header = None)
transactions = []
for i in range(0, 7501):
    #串列生成式
    transactions.append([str(data.iloc[i, j]) for j in range(0, 20)])
#先驗演算
from apyori import apriori
rules = apriori(transactions, min_support = 0.003,
        min_confidence = 0.2, min_lift = 3, min_length = 2)

#視覺化
results = list(rules)
myresults = [list(x) for x in results] 